lib/srv/db/sqlserver/protocol/login7.go

```diff
@@ -143,3 +140,44 @@ func ReadLogin7Packet(r io.Reader) (*Login7Packet, error) {
 		database: database,
 	}, nil
 }
+
+// errInvalidPackage is returned when Login7 package contains invalid data.
+var errInvalidPackage = trace.Errorf("invalid login7 packet")
+
+// readUsername reads username from login7 package.
+func readUsername(pkt *Packet, header Login7Header) (string, error) {
+	if len(pkt.Data) < int(header.IbUserName)+int(header.CchUserName)*2 {
+		return "", errInvalidPackage
+	}
+
+	if len(pkt.Data) <= int(header.IbUserName) {
+		return "", errInvalidPackage
+	}
```

Comment:
Isn't this condition already covered by the check above? Unless `CchUserName` is somehow negative...

Reply:
You're right. I remamber that at some point I was getting a negative value, but after some cleanup now this if is not needed anymore and `CchUserName` is `uint16`, so it will never be negative. Re-tested and removed.